### PR TITLE
Issue 19029: Initial code for PKCE support in OIDC RP

### PR DESCRIPTION
**Pull request by ayoho**
Should be the bulk of the code needed to prototype PKCE. The caching functionality still could use another look.

For #19029

### CODE REVIEW

**Comment by ayoho** [comment]
#build

**Review by arunavemulapalli — DISMISSED** [review]
lgtm

File: dev/com.ibm.ws.security.oauth/bnd.bnd [hunk 1]
```diff
@@ -145,7 +142,6 @@ Include-Resource: \
 	com.ibm.ws.webcontainer.security;version=latest,\
 	com.ibm.ws.webcontainer;version=latest,\
 	com.ibm.wsspi.org.osgi.service.component.annotations;version=latest,\
-	com.ibm.ws.security.openidconnect.clients.common;version=latest,\
```
Comment (arunavemulapalli) [inline dev/com.ibm.ws.security.oauth/bnd.bnd:148]: I can't remember why we had this build dependency

**Review by jimmy1wu — DISMISSED** [review]

**Comment by ayoho** [comment]
#libby
#build

**Comment by ayoho** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oauth_fat,com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.OpenShift,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.okdServiceLogin,io.openliberty.security.jakartasec.3.0.internal_fat,io.openliberty.security.jakartasec.3.0.internal_fat.config.1,io.openliberty.security.jakartasec.3.0.internal_fat.config.2,io.openliberty.security.jakartasec.3.0.internal_fat.logout,io.openliberty.security.jakartasec.3.0.internal_fat.refresh

**Review by jimmy1wu — DISMISSED** [review]

**Comment by ayoho** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oauth_fat,com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.OpenShift,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.okdServiceLogin,io.openliberty.security.jakartasec.3.0.internal_fat,io.openliberty.security.jakartasec.3.0.internal_fat.config.1,io.openliberty.security.jakartasec.3.0.internal_fat.config.2,io.openliberty.security.jakartasec.3.0.internal_fat.logout,io.openliberty.security.jakartasec.3.0.internal_fat.refresh

**Review by c00crane — APPROVED** [review]

**Pull request merged**